packages/babel-generator/test/index.js

```diff
@@ -914,6 +917,61 @@ describe("generation", function () {
       }
     `);
   });
+
+  it("should not throw when loc.column === 0 with inputSourceMap", () => {
+    const ast = parseExpression("a(\n)");
+
+    ast.loc.end.column = 0;
```

Comment:
Can we construct a test without AST location mutations? If not it may point to an upstream issue and we can provide a better error message.

The current `_normalizePosition` implementation will also fail if any external tool sets `ast.loc.end.line` to `-1`. If we decide to mute such errors, we should also handle the `.line` info.

Reply:
This happened in a complex Angular project, and I thought it was a downstream bug, but since it's a regression, I took this approach.

> The current _normalizePosition implementation will also throw if any external tool sets ast.loc.end.line to -1. If we decide to mute such errors, we should also handle the .line info.

You are right, but the reason for this regression is that we internally added `columnOffset` resulting in a possible -1, and `lineOffset` is not currently actually used, I'm not sure about doing the same.
Maybe we can remove `lineOffset`?

Reply:
Anything new here ? It would be awesome to have this fix :)